cirq-core/cirq/ops/gateset.py

```diff
@@ -280,18 +299,20 @@ def val_if_none(var: Any, val: Any) -> Any:
 
         name = val_if_none(name, self._name)
         unroll_circuit_op = val_if_none(unroll_circuit_op, self._unroll_circuit_op)
-        accept_global_phase_op = val_if_none(accept_global_phase_op, self._accept_global_phase_op)
+        global_phase_family = GateFamily(gate=global_phase_op.GlobalPhaseGate)
         if (
             name == self._name
             and unroll_circuit_op == self._unroll_circuit_op
-            and accept_global_phase_op == self._accept_global_phase_op
+            and (not accept_global_phase_op or global_phase_family in self.gates)
```

Comment:
I considered that but then that would cause the following weird behavior:

```python
orig = cirq.Gateset(cirq.X, accept_global_phase_op=True)
x1 = orig.with_params(accept_global_phase_op=False)
x2 = cirq.Gateset(*orig.gates, accept_global_phase_op=False)
assert x1 == x2  # fails
```

Reply:
Fixed. I also made the parameter in the constructor to `Optional[bool]` to make it consistent.